src/main/java/org/opensearch/agent/tools/PPLTool.java

```diff
@@ -285,7 +285,9 @@ private void extractNamesTypes(Map<String, Object> mappingSource, Map<String, St
             if (v instanceof Map) {
                 Map<String, Object> vMap = (Map<String, Object>) v;
                 if (vMap.containsKey("type")) {
-                    fieldsToType.put(prefix + n, (String) vMap.get("type"));
+                    if (!((vMap.get("type")).equals("alias"))) {
```

Comment:
Is it possible that vMap.get("type") returns null?

Reply:
I don't think so. Every field inside our mapping should have a field. But I change it to getOrDefault for safety.